dss/storage/checkout/file.py

```diff
@@ -17,8 +17,8 @@ def start_file_checkout(replica: Replica, blob_key, dst_bucket: typing.Optional[
                        for the replica.
     :return: The execution ID of the request.
     """
-    if dst_bucket is None:
-        dst_bucket = replica.checkout_bucket
+    dst_bucket = dst_bucket or replica.checkout_bucket
+    assert dst_bucket != replica.bucket, f'Cannot checkout a file from {dst_bucket} to itself!'
```

Comment:
Please add a comment indicating these lines should be removed after the checkout misconfiguration has been identified and fixed.